sentinelsat/advanced.py

```diff
@@ -0,0 +1,359 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, division
+
+import os
+import shutil
+from xml.etree import ElementTree as etree
+
+import sentinelsat
+from sentinelsat.sentinel import InvalidChecksumError, _check_scihub_response
+from sentinelsat.exceptions import SentinelAPIError
+
+
+def _xml_to_dataobj_info(element):
+    assert etree.iselement(element)
+    assert element.tag == "dataObject"
+    data = dict(
+        id=element.attrib["ID"],
+        rep_id=element.attrib["repID"],
+    )
+    elem = element.find("byteStream")
+    # data["mime_type"] = elem.attrib['mimeType']
+    data["size"] = int(elem.attrib["size"])
+    elem = element.find("byteStream/fileLocation")
+    data["href"] = elem.attrib["href"]
+    # data['locator_type'] = elem.attrib["locatorType"]
+    # assert data['locator_type'] == "URL"
+
+    elem = element.find("byteStream/checksum")
+    assert elem.attrib["checksumName"].upper() == "MD5"
+    data["md5"] = elem.text
+
+    return data
+
+
+class AdvancedSentinelAPI(sentinelsat.SentinelAPI):
```

Comment:
What does this class inherit from `SentinelAPI` that is still used, apart from the init that stores the credentials? Will class methods from `SentinelAPI` that are not overwritten still work on this class, like `download_all`, `query`, etc and is it intended that these (and all the others) are still found on this child class? If we only need very little common functionality, perhaps we should introduce a common parent class (perhaps ABC) instead - or just copy the relevant part of the interface.

Reply:
Yes the idea is that the new class just *add* some extra feature to the ones already implemented in the standard ``SentinelAPI`` (see also the description in the ``api_overview.rst``).

This allows to replace ``SentinelAPI`` with ``AdvancedSentinelAPI`` (or whatever new name we decide) in the ``cli()`` function without adding too much additional logic on the client side.